File: PUPro/src/piCode/ComfZone.java
package piCode;

import sensorAccess;

public class ComfZone {
	
	private Pin rightPin;
	private Pin lefPin;
	private boolean on;
	private double sensorDataLeft;
	private double sensorDataRight;
	
	public final double safeDistance = 2; 
	
	public void checkZone(){
		
		// placeholder
		sensorDataLeft = getSensorData();
		sensorDataRight = getSensorData();
		
		while (on==true){
			if (checkLeft(sensorDataLeft)) {
				leftPin.toggleOn; //placeholder
			} else {
				leftPin.toggleOff;
			}
			
			if (checkRight(sensorDataRight)) {
				rightPin.toggleOn;
			} else {
				rightPin.toggleOff;
			}
		}
		
	}
	
	private boolean checkLeft(double sensorData) {
		if (sensorData < safeDistance) {
			return true;
		}
		return false;
	}

	private boolean checkRight(double sensorData) {
		if (sensorData < safeDistance) {
			return true;
		}
		return false;
	}
	
	public boolean isOn() {
		return on;
	}

	public void setOn(boolean on) {
		this.on = on;
	}
	
	public static void main(String[]args){
		//Main-method
	}
	
}

File: chr/comfZone.py
import sensorAccess

class ComfZone():
    sensorDataLeft = sensorAccess.get() #placeholder function, should get data
    sensorDataRight = sensorAccess.get()
    isOn = False #whether ComfZone is turned on by user
    leftPin,rightPin #for giving feedback on whether a car is in your blindzone

    safeDistance = 8#final num to compare with distance from sensor

    def checkZone():
        while (isON):
            sensorDataLeft = sensorAccess.get()
            sensorDataRight = sensorAccess.get()

            if (check(sensorDataLeft)):
                leftPin.toggleOn
            else:
                leftPin.toggleOff

            if (check(sensorDataRight)):
                rightPin.toggleOn
            else:
                rightPin.toggleOff


    def check(sensorData):
        if (sensorData < safeDistance):
            return true
        return false

    def setOn(on):
        self.isOn = on
